server/commands/list_objects.go

```diff
@@ -0,0 +1,232 @@
+		listObjectsMaxResults:        listObjectsMaxResults,
+		resolveNodeLimit:             ResolveNodeLimit,
+	}
+}
+
+// Execute the ListObjectsQuery, returning a list of object IDs
+func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjectsRequest) (*openfgapb.ListObjectsResponse, error) {
+	err := q.validateInput(ctx, req.StoreId, req.Type, req.AuthorizationModelId, req.Relation)
+	if err != nil {
+		return nil, err
+	}
+
+	resultsChan := make(chan string, q.listObjectsMaxResults)
+	errChan := make(chan error)
+	resolvedChan := make(chan struct{})
+
+	timeoutCtx := ctx
+	var cancel context.CancelFunc
+	if q.listObjectsDeadlineInSeconds != 0 {
+		timeoutCtx, cancel = context.WithTimeout(ctx, time.Second*time.Duration(q.listObjectsDeadlineInSeconds))
+		defer cancel()
+	}
+
+	go func() {
+		q.performChecks(timeoutCtx, req.StoreId, req.AuthorizationModelId, req.Type, req.Relation, req.User, req.ContextualTuples, resultsChan, errChan, resolvedChan)
+	}()
+
+	select {
+	case <-timeoutCtx.Done():
+		return nil, serverErrors.NewInternalError("Timeout exceeded", timeoutCtx.Err())
+	case <-resolvedChan:
+		objectIDs := make([]string, 0)
+		for objectID := range resultsChan {
+			objectIDs = append(objectIDs, objectID)
+		}
+
+		return &openfgapb.ListObjectsResponse{
+			ObjectIds: objectIDs,
+		}, nil
+	case genericError := <-errChan:
+		return nil, genericError
+	}
+}
+
+func (q *ListObjectsQuery) ExecuteStreamed(ctx context.Context, req *openfgapb.StreamedListObjectsRequest, srv openfgapb.OpenFGAService_StreamedListObjectsServer) error {
+	err := q.validateInput(ctx, req.StoreId, req.Type, req.AuthorizationModelId, req.Relation)
+	if err != nil {
+		return err
+	}
+
+	resultsChan := make(chan string, q.listObjectsMaxResults)
+	errChan := make(chan error)
+	resolvedChan := make(chan struct{})
+
+	timeoutCtx := ctx
+	var cancel context.CancelFunc
+	if q.listObjectsDeadlineInSeconds != 0 {
+		timeoutCtx, cancel = context.WithTimeout(ctx, time.Second*time.Duration(q.listObjectsDeadlineInSeconds))
+		defer cancel()
+	}
+
+	go func() {
+		q.performChecks(timeoutCtx, req.StoreId, req.AuthorizationModelId, req.Type, req.Relation, req.User, req.ContextualTuples, resultsChan, errChan, resolvedChan)
+	}()
+
+	for {
+		select {
+		case <-timeoutCtx.Done():
+			return serverErrors.NewInternalError("Timeout exceeded", timeoutCtx.Err())
+		case objectID, ok := <-resultsChan:
+			if ok {
+				err := srv.Send(&openfgapb.StreamedListObjectsResponse{
+					ObjectId: objectID,
+				})
+				if err != nil {
+					return serverErrors.NewInternalError("", err)
+				}
+			} else {
+				// the channel was closed
+				return nil
+			}
+		case genericError, ok := <-errChan:
+			if ok {
+				return serverErrors.NewInternalError("", genericError)
+			}
+		}
+	}
+}
+
+func (q *ListObjectsQuery) getUniqueObjects(ctx context.Context, storeID, targetObjectType string, ctxTuples *openfgapb.ContextualTupleKeys) ([]string, error) {
```

Comment:
Do we need `Unique` in the name?

Reply:
![image](https://user-images.githubusercontent.com/5374887/183544373-0b0d4b3b-abab-49fb-8bb1-4adf1d4cc803.png)